SilKit/source/services/orchestration/TimeSyncService.cpp

```diff
@@ -546,6 +612,75 @@ bool TimeSyncService::AbortHopOnForCoordinatedParticipants() const
     }
     return false;
 }
+
+auto TimeSyncService::GetCurrentWallClockSyncPoint() const -> std::chrono::nanoseconds
+{
+    Lock lock{_mx};
+    return _currentWallClockSyncPoint;
+}
+
+void TimeSyncService::StartWallClockCouplingThread()
+{
+    _wallClockCouplingThreadRunning = true;
+    _stopFuture = _wallClockCouplingThreadStopPromise.get_future();
+
+    _wallClockCouplingThread = std::thread{[&]() {
+
+        SilKit::Util::SetThreadName("SilKit-WallClockCoupling");
+
+        auto startTime = std::chrono::steady_clock::now();
+        const auto duration = _timeConfiguration.NextSimStep().duration;
```

Comment:
This value might not be constant across the simulation. If the user sets the simulation step handler mutliple times, they can specify different durations each time.

Reply:
Ok, I directly get them from `_timeConfiguration.NextSimStep().duration`